src/modules/quick_api_module.f90

```diff
@@ -338,48 +328,107 @@ subroutine get_atom_types(natoms, atomic_numbers, natm_type, atm_type_id, ierr)
 
 end subroutine get_atom_types
 
-! returns quick qm energy
-subroutine get_quick_energy(coords, ptchg_crd, energy, ierr)
+! allocate memory for point charges and gradients
+subroutine allocate_point_charge(isgrad,ierr)
+
+  use quick_molspec_module
+  use quick_calculated_module
+  implicit none
+  logical, intent(in) :: isgrad
+  integer, intent(inout) :: ierr
+  
+  ! allocate memory only if external charges exist
+  if ( .not. allocated(quick_api%ptchg_crd)) allocate(quick_api%ptchg_crd(4,quick_api%nxt_ptchg), stat=ierr)
+
+  call realloc(quick_molspec,ierr)
+
+  if(isgrad) then
+    if ( .not. allocated(quick_api%ptchg_grad)) allocate(quick_api%ptchg_grad(3,quick_api%nxt_ptchg), stat=ierr)
+    quick_api%ptchg_grad =0.0d0
+    call realloc(quick_qm_struct,ierr)
```

Comment:
It seems like you need that `call realloc()` only once if you put it outside of the `if(isgrad)` statement at the end of the routine.

Reply:
Oh scratch that. Those are two different structs. I will delete my comment.